Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3026,6 +3095,68 @@ public void run() {
 		});
 	}
 
+	/*
+    * Checks if the BranchActivityLifeCycleObserver is initialised
+    * @return true if BranchActivityLifeCycleObserver initialised else false
+    */
+	public boolean isActivityObserverInitialised(){
+		return activityLifeCycleObserver_ != null;
+	}
+
+	@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+	private void setActivityLifeCycleObserver(Application application){
+		application.registerActivityLifecycleCallbacks(new BranchActivityLifeCycleObserver());
+	}
+
+	/**
+	 * <p>Class that observes activity life cycle events and determines when to start and stop
+	 * session.</p>
+	 */
+	@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+	private class BranchActivityLifeCycleObserver implements Application.ActivityLifecycleCallbacks{
+		int m_ActivityCnt = 0; //Keep the count of live  activities
```

Comment:
Can we stick with the private variable naming convention that we use elsewhere. Eg:
private boolean initNotStarted_;